edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -350,13 +354,32 @@ impl ModuleRuntime for DockerModuleRuntime {
                     module.config().image()
                 );
 
-                let create_options = create_options
+                let mut create_options = create_options
                     .with_image(module.config().image().to_string())
                     .with_env(merged_env)
                     .with_labels(labels);
 
-                // Here we don't add the container to the iot edge docker network as the edge-agent is expected to do that.
-                // It contains the logic to add a container to the iot edge network only if a network is not already specified.
+                // In the bootstrap scenario (edge runtime module starting up for the first time),
+                // add the container to the custom network. We identify this scenario by looking
+                // for 'NetworkId' in the module's environment. The edge runtime module is then
+                // responsible for adding all subsequent modules to the custom network.
+                if let Some(network_id) = module.env().get("NetworkId") {
+                    let mut network_config = create_options
+                        .networking_config()
+                        .and_then(|network_config| serde_clone(network_config).ok())
```

Comment:
What does `serde_clone` do that a normal `.clone()` wouldn't?

Reply:
Frankly not sure. This was a copy/paste of the original logic that was removed by e2006e0 (and replaced with the comment "Here we don't add the container to the iot edge docker network..."). Maybe @varunpuranik remembers?

Reply:
Yeah, I'd spelunked yesterday and found it was originally added in https://github.com/Azure/iotedge/commit/aa2740644eba812bb96c0ee0299f97296db719e6 , but that doesn't explain either.

Reply:
I checked with Raj. He says it may have been because he didn't want to add `Clone` to the `docker::models::*` types by hand. But I saw Varun added them by hand in https://github.com/Azure/iotedge/commit/2cbe1bd74109fad7a1fc69398de2218eae423d44 anyway, so that doesn't matter any more.

I think we should `Clone` instead of `serde_clone`, at least for the "new" code in this PR. Though if you remove `serde_clone` entirely in this PR I'll support it. The only other use is for `ContainerCreateBody` which is already `Clone` from Varun's change.